analyzers/src/SonarAnalyzer.Common/SyntaxTrackers/ObjectCreationTracker.cs

```diff
@@ -21,82 +21,56 @@
+                       && context.InvokedConstructorSymbol.Value.Parameters.Length > index
+                       && context.InvokedConstructorSymbol.Value.Parameters[index].Type.Is(type);
 
         internal ObjectCreationCondition WhenDerivesOrImplementsAny(params KnownType[] types) =>
             context => context.InvokedConstructorSymbol.Value != null
-                    && context.InvokedConstructorSymbol.Value.IsConstructor()
-                    && context.InvokedConstructorSymbol.Value.ContainingType.DerivesOrImplementsAny(types.ToImmutableArray());
+                       && context.InvokedConstructorSymbol.Value.IsConstructor()
+                       && context.InvokedConstructorSymbol.Value.ContainingType.DerivesOrImplementsAny(types.ToImmutableArray());
 
         internal ObjectCreationCondition MatchConstructor(params KnownType[] types) =>
             // We cannot do a syntax check first because a type name can be aliased with
             // a using Alias = Fully.Qualified.Name and we will generate false negative
             // for new Alias()
             context => context.InvokedConstructorSymbol.Value != null
-                    && context.InvokedConstructorSymbol.Value.IsConstructor()
-                    && context.InvokedConstructorSymbol.Value.ContainingType.IsAny(types);
+                       && context.InvokedConstructorSymbol.Value.IsConstructor()
+                       && context.InvokedConstructorSymbol.Value.ContainingType.IsAny(types);
 
         internal ObjectCreationCondition WhenDerivesFrom(KnownType baseType) =>
             context => context.InvokedConstructorSymbol.Value != null
-                    && context.InvokedConstructorSymbol.Value.IsConstructor()
-                    && context.InvokedConstructorSymbol.Value.ContainingType.DerivesFrom(baseType);
+                       && context.InvokedConstructorSymbol.Value.IsConstructor()
+                       && context.InvokedConstructorSymbol.Value.ContainingType.DerivesFrom(baseType);
 
         internal ObjectCreationCondition WhenImplements(KnownType baseType) =>
             context => context.InvokedConstructorSymbol.Value != null
-                    && context.InvokedConstructorSymbol.Value.IsConstructor()
-                    && context.InvokedConstructorSymbol.Value.ContainingType.Implements(baseType);
+                       && context.InvokedConstructorSymbol.Value.IsConstructor()
+                       && context.InvokedConstructorSymbol.Value.ContainingType.Implements(baseType);
+
+        protected override BaseContext IsTracked(SyntaxNode expression, SemanticModel semanticModel, ObjectCreationCondition[] conditions, out Location location)
+        {
+            var context = new ObjectCreationContext(expression, semanticModel);
+            location = expression.GetLocation();
+            return conditions.All(c => c(context)) ? context : null;
```

Comment:
This repeats a lot.
The location should be fixed by another comments. 
Can you do a little bit of generic magic between context, condition and tracker to be able to move the last line in `SyntaxTrackerBase`?
That would allow this method to be just `CreateContext` and have very clean design.